STEP_Project/content.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Function that will later have Add/Remove functionality
+ * @param {JS event} event - expect click event
+ */
+function arButtonClick (event) 
+{
+    if (!event.target.matches('BUTTON')) return
+    alert("hello");
+}
+
+/**
+ * This adds a button to every row in the table
+ * If the row already has a button, it skips that row
+ * @param {HTML Object} OUs - List of OUs
+ * @param {integer} numRows - Number of rows
+ */
+function addButtons(OUs, numRows)
+{
+    for (let i = 1; i < numRows; i++) 
+    {
+        let row = OUs[i];
+        let temp = row.getElementsByClassName('bClass');
+        if (temp.length == 0)
+        {
+            const button = document.createElement('BUTTON');
+            button.innerHTML = 'click me';
+            button.setAttribute('class','bClass');
+            row = OUs[i];
+            row.appendChild(button);
```

Comment:
1) Table rows (tr) should only have table cell (td) as a child element.
https://www.w3.org/TR/html401/struct/tables.html

2) Hide existing buttons, otherwise you will catch their clicks also.

```
// Create a new table cell element with button
const cell = document.createElement('TD');
cell.style.position = 'absolute';
cell.style.right = '10px';
cell.appendChild(button);

// Hide existing cell with actions
row.lastChild.style.opacity = 0;
row.lastChild.style.pointerEvents = 'none';

// Append the new cell to the row
row.appendChild(cell);
```